test/rules/indent/spaces-2/test.ts.lint

```diff
@@ -159,4 +159,9 @@ var arr2 = [
   }
 ];
 
+var arr3 = [
+    1
+~~~~ [0]
+];
```

Comment:
this needs another test for nesting:
```ts
var arr3 = [
    {
        foo,
    }
]
```

Also add a test with more than 4 spaces indent:
```ts
var foo =
          bar;
```

Reply:
As for the "another" `var arr3 = [` case, this(fixed) implementation cannot fix it correctly due to, there isn't some kind of multiline-ready fix. So, after the `--fix` this example will look like:
```js
var arr3 = [
  {
    foo,
    }
]
```
The latest `}` won't be treated by the tslint as an issue due to, it conforms naive rules of this(fixed) implementation.
To fix that issue correctly, we should dig into the AST of a linting code fragment much deeper. Otherwise, a naive implementation won't feel the difference between your example and a single-line code-style error, e.g.:
```js
var arr3 = [
    { // the error is only here
    foo,
  }
]
```
which may be fixed by a naive multiline-ready fixer as:
```js
var arr3 = [
  { // the error is only here
  foo,
}
]
```

So, seems like we'd better not fix indent size violations automatically. Instead of it, we should only report a notification and rely on only a manual fixing in these cases.